sway/input/cursor.c

```diff
@@ -822,6 +822,7 @@ void dispatch_cursor_button(struct sway_cursor *cursor,
 	// Handle existing seat operation
 	if (cursor->seat->operation != OP_NONE) {
 		if (button == cursor->seat->op_button && state == WLR_BUTTON_RELEASED) {
+			state_erase_button(cursor, button);
```

Comment:
Won't this still go into an unexpected state if I press or release another button during a seat operation? By another button I mean a button that isn't `cursor->seat->op_button`.

For example:

    if (button == cursor->seat->op_button && state == WLR_BUTTON_RELEASED) {
        ...
    }
    if (state == WLR_BUTTON_PRESSED) {
        state_add_button(cursor, button);
    } else {
        state_erase_button(cursor, button);
    }
    return;

Reply:
Actually, it wouldn't since the state just wouldn't change during an operation and state is only used for bindings, but I think this is a better approach anyway.